File: 8kyu/speed-code-2-array-madness/speed-code-2-array-madness.py
# def array_madness(a,b):
#   def square(arr, pow):
#     return [num ** pow for num in arr]
  
#   return sum(square(a, 2)) > sum(square(b, 3))

def array_madness(a,b):
  return sum(num ** 2 for num in a) > sum(num ** 3 for num in b)


print(array_madness([4, 5, 6], [1, 2, 3]))

File: 8kyu/speed-code-2-array-madness/speed-code-2-array-madness.js
/* 
PREP:
- set loopCalc fuction
  - power of 2/3 for a/b
  - get sum for both arrays
- return true if a is greater than b
*/

function arrayMadness(a, b) {
	function loopCalc(array, power) {
		return array.map((num) => num ** power).reduce((acc, cur) => acc + cur, 0);
	}
	return loopCalc(a, 2) > loopCalc(b, 3);
}
